### DISCUSSION TITLE
Max size of Hamiltonian for mcsolve

### DISCUSSION THREAD

**Discussion by abhinsuresh** [qutip/qutip#2250] [Q&A]
1. I would like to know the largest matrix size the mcsolve can handle. I want to do the quantum trajectory method for a system of size 65660 (H and psi are sparse but L is not). I have made all the required matrix to Qobj, but when passing them to mcsolve, it takes too much time to start the solver. 
Will using Option(rhs_reuse) will help the calculations to start faster?
2. For solving the spin system using the universal Lindblad equation, evolving the system using (I - Heff*dt)|psi> the norm of wave function increases with time step, so my quantum trajectory code fails to produce the correct results. Qutip still produces correct results. if quitp follows the simple algorithm mentioned in (https://qutip.org/docs/latest/guide/dynamics/dynamics-monte.html) qutip method should also fail. Am I missing something?

**Comment by Ericgig** [comment]
1. If your RAM can support it, matrix size can go up to 2**31-1. You can try `rhs_reuse` but I don't expect it to be that much faster and could cause errors if not used properly. With large, not that sparse matrices computation are slow and there will be no feedback until 10% of the trajectories are finished.
2. You are probably missing something. With mcsolve, the norm decrease with time until a jump where it is renormalized.
